java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -94,10 +95,41 @@ public JObject(
       for (Map.Entry<String, JSONSchemaProps> field : fields.entrySet()) {
         String nextPrefix = (config.getPrefixStrategy() == Config.Prefix.ALWAYS) ? classPrefix : "";
         String nextSuffix = (config.getSuffixStrategy() == Config.Suffix.ALWAYS) ? classSuffix : "";
+        String fieldKey = field.getKey();
+        // let's handle duplicates
+        final Map<String, Map<String, JSONSchemaProps>> duplicates = new HashMap<>();
+        // counting the current field duplicates
+        final List<Map.Entry<String, JSONSchemaProps>> fieldDuplicates = fields.entrySet().stream()
```

Comment:
Despite the fact that I really like the readability of this solution it involves, at the moment, a pretty expensive computation.
Have you attempted to perform the field name check/sanitization once per `JObject` before the loop iterating on the fields (e.g. before line 95)?